### PR TITLE
Add .env file and symlinks

### LINKED ISSUE #10: Initialize .env file
# Overview
have global .env file with symlinks to frontend and backend

There are several variables that we will want to use as environment variables since we won't want to commit them to the git history. To do this, we will want to use a `.env` file. 

Since the project is split into a frontend and a backend, we will need both to be able to reference the `.env` file which is typically at the root level of the project, which for the frontend would be `frontend/.env` and for the backend it would be `backend/.env`.

The end result should look like:
```bash
/ 
frontend/
  .env (symlink)
backend/
  .env (symlink)
.env
```

This means we can just update the one `.env` in the root directory of the project, and the frontend and backend apps can access that via a symlink.

To verify the functionality, we can do a simple console.log or print statement of a env var in each app. 

### Recommended Steps
1. Create a .env file in frontend and backend to test accessing env vars
2. refactor into the single .env file and replace the frontend and backend .env files with symlinks

Ref: https://www.futurelearn.com/info/courses/linux-for-bioinformatics/0/steps/201767#:~:text=A%20symlink%20is%20a%20symbolic,directory%20in%20any%20file%20system.

### PR DESCRIPTION
**Pull request by DAPotapov**
## Overview
Closes #10 
Added .env file with content for backend only. Added test for it. Added symlinks to it in backend and frontend directories. Updated requirements.txt to use dotenv.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: backend/test_app.py [hunk 1]
```diff
@@ -22,7 +23,7 @@ def test_get_home(client):
 def test_get_recipe(client):
     resp = client.get("/recipe")
     assert (
-        b'{"author":"John Doe","description":"A delicious pizza!","title":"Pizza"}\n'
+        b'{\n  "author": "John Doe", \n  "description": "A delicious pizza!", \n  "title": "Pizza"\n}\n'
```
Comment (MSevey) [inline backend/test_app.py:26]: I think this was a mistake, this will break the test

Comment (DAPotapov) [reply backend/test_app.py:26]: Actually when I was solving issue #10 and run backend-test (without touching these lines, they were as pulled) I got assertion errors for these tests:
`E       assert b'{"author":"John Doe","description":"A delicious pizza!","title":"Pizza"}\n' in b'{\n  "author": "John Doe", \n  "description": "A delicious pizza!", \n  "title": "Pizza"\n}\n'
E        +  where b'{\n  "author": "John Doe", \n  "description": "A delicious pizza!", \n  "title": "Pizza"\n}\n' = <WrapperTestResponse 88 bytes [200 OK]>.data

test_app.py:25: AssertionError` 
I tried to figure out whats wrong but in the end just copypasted what pytest expect to see.
I don't like it either, but backend-test passed after that. I propose to refactor these lines in a manner like 'assert author in  resp.data AND title in resp.data'. Something like that. But I think it should be separated PR.

Comment (DAPotapov) [reply backend/test_app.py:26]: Weird: only these lines passed local test, but failed test on github. Maybe it's because of different versions of python3.

Comment (DAPotapov) [reply backend/test_app.py]: Test passed

File: backend/test_app.py [hunk 2]
```diff
@@ -48,6 +49,12 @@ def test_post_recipe(client):
         },
     )
     assert (
-        b'{"author":"babushka","description":"Meatballs in dough","title":"Pelmeny"}\n'
+        b'{\n  "author": "babushka", \n  "description": "Meatballs in dough", \n  "title": "Pelmeny"\n}\n'
```
[the remark this answers is not in the document]
Comment (DAPotapov) [reply backend/test_app.py]: Test passed

File: .env [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Backend environment variables
+FLASK_APP = app.py
+FLASK_DEBUG = true
+
+# Frontend environment variables
+# REACT_APP_?? = ??
```
Comment (MSevey) [inline .env:6]:
```suggestion
# REACT_APP_TEST_ENV = "my variable"
```
Let's add a simple console.log statement in the frontend to verify that these variables can be accessed as well

Comment (DAPotapov) [reply .env:6]: About versions:
I installed separate environment with Python 3.11 using venv, but every time I run:` make backend-test` - it uses version installed by default in my system, which is:
```
platform linux -- Python 3.10.6, pytest-7.2.2, pluggy-0.13.0 -- /usr/bin/python3
cachedir: .pytest_cache
plugins: doctestplus-0.11.2, filter-subpackage-0.1.1, mock-3.6.1, astropy-header-0.2.0, cov-3.0.0, remotedata-0.3.3, openfiles-0.5.0, arraydiff-0.5.0, hypothesis-6.36.0
```
(Is it normal? Maybe conda is better choice?)
By the way, test on github passed, but it uses Python 3.11 and more recent libraries than mine...

**Review by MSevey — APPROVED** [review]

**Pull request merged**